python/ray/tune/suggest/bayesopt.py

```diff
@@ -111,21 +118,31 @@ def __init__(self,
             self._metric_op = 1.
         elif mode == "min":
             self._metric_op = -1.
+
         self._live_trial_mapping = {}
+        self._cached_results = []
+        self._random_search_steps = random_search_steps
+        self._random_search_trials = 0
 
         self.optimizer = byo.BayesianOptimization(
             f=None, pbounds=space, verbose=verbose, random_state=random_state)
 
         self.utility = byo.UtilityFunction(**utility_kwargs)
+
+        # Registering the provided analysis, if given
         if analysis is not None:
             self.register_analysis(analysis)
 
     def suggest(self, trial_id):
         if self.max_concurrent:
             if len(self._live_trial_mapping) >= self.max_concurrent:
                 return None
-        new_trial = self.optimizer.suggest(self.utility)
+        if self._random_search_steps > 0 and len(self._cached_results) < self._random_search_steps:
```

Comment:
I think this will fail the lint test because it is too long

Reply:
Yes, at the current time I have restored it to a previous version since I needed to run a test and it had some bug on the penultimate update. I should have everything fixed by this evening.